### ISSUE TITLE
Go to previous/next member

### ISSUE THREAD

**Issue by MarkZ3** [theia-ide/theia#3628]
In Eclipse, there was a command in the editor to go to previous/next member (Alt+shift+up/down or ctrl+option+up/down). For Java and C++, it pretty much just navigated the outline view in the same order as seen in the file. Perhaps the result of document/symbol can be reused for this. This is also not present in vscode BTW.

Why is this useful? Sometimes it's faster to go up or down a couple of types than using the "go to symbol in file" and typing the name. It also can be useful when you're in the middle of a method body and want to jump to its signature.

**Comment by kittaakos** [comment]
Did you mean something like this https://marketplace.visualstudio.com/items?itemName=mishkinf.goto-next-previous-member?

According to the description, it works with the `DocumentSymbol.range`, received from the LSP:

> For the outline to work, the language support plugins need to support symbol information.
> 
> For the outline to form a tree structure, the language support plugins need to report the > entire definition range as part of symbol.
>
> See VS Code issue #34968 and language server protocol issue #132 for a discussion.

I think we could do the same, directly from the outline, or by getting the symbols (`textDocument/documentSymbol`) from the LS.

**Comment by MarkZ3** [comment]
Yes, it looks the same as the Eclipse feature. Would it make more sense that I create an issue in Monaco so that the features ends up in both vscode and Theia? Would that be the right component or something else?
